File: 66-PlusOne/66-PlusOne.cpp
// Last updated: 08/05/2026, 09:32:59
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        int carry=0;
        int n= digits.size();
        for(int i=n-1;i>=0;i--){
            if(digits[i]<9 && i==n-1){
                digits[i]+=1;
                break;
            }
            else if(carry==1 && digits[i]<9){
                digits[i]+=1;
                carry=0;
                break;
            }
            else if(carry==1 && digits[i]==9){
                digits[i]=0;
            }
            else if(digits[i]==9 && i==n-1)
            {digits[i]=0;
            carry=1;
            }
        }
        if (carry==1){
            digits.insert(digits.begin(),1);
        }
        return digits;


        }
};

File: 66-PlusOne/66-PlusOne.py
# Last updated: 08/05/2026, 09:32:55
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        n= len(digits)
        for i in range(n-1,-1,-1):
            if i==n-1:
                digits[i]+=1
            
            if digits[i]==10:
                if i==0:
                    digits[i]=1
                    digits.append(0)
                else:
                    digits[i]=0
                    digits[i-1]+=1
                
            else:
                break

        return digits

        